cloud-controller-manager/civo/loadbalancer.go

```diff
@@ -249,23 +249,36 @@ func getLoadBalancer(ctx context.Context, c *civogo.Client, kclient kubernetes.I
 	} else if name, ok := service.Annotations[annotationCivoLoadBalancerName]; ok {
 		civolb, err = c.GetLoadBalancer(name)
 	} else {
-		lbName := getLoadBalancerName(clusterName, service)
-		civolb, err = c.FindLoadBalancer(lbName)
+		cluster, cerr := c.GetKubernetesCluster(ClusterID)
+		if cerr != nil {
+			klog.Errorf("Unable to get kubernetes cluster, error: %v", err)
+			return nil, err
+		}
+		lbName := getLoadBalancerName(cluster.Name, service)
+		civolb, err = c.GetLoadBalancer(lbName)
 		if err == nil {
 			patcher := newServicePatcher(kclient, service)
 			defer func() { err = patcher.Patch(ctx, err) }()
 
 			updateServiceAnnotation(service, annotationCivoLoadBalancerID, civolb.ID)
 			updateServiceAnnotation(service, annotationCivoLoadBalancerName, civolb.Name)
+			updateServiceAnnotation(service, annotationCivoClusterID, ClusterID)
```

Comment:
Can we make a struct with all the annotations?

`type lbmeta struct{
ID string 
Name string
// so on
}`
And then we can have  `updateServiceAnnotation(service , lbmeta)`

Reply:
Yes, we can but not in this PR